Compilation fails with --disable-werror

When building lxc with --disable-werror, gnu11 is not added in CLFAGS, moreover lxc does not add gnu99 so compilation fails on:
```
In file included from ./utils.h:48:0,
                 from cgroups/cgfsng.c:59:
cgroups/cgfsng.c: In function 'lxc_cpumask':
./macro.h:135:2: error: 'for' loop initial declarations are only allowed in C99 or C11 mode
  for (char *__p = NULL, *__it = strtok_r(__splitme, __separators, &__p)
```

I would be happy to send a PR to add gnu99 if werror is disabled but I'm not sure that this is what is intended. Indeed, I found conflicting information on the minimum gcc version required by lxc:
 - In the commit message https://github.com/lxc/lxc/commit/81a56e8029e5c6815ed3ac9835a1d3822a22dae0, it is mentioned "We can't really support anything less than **gcc-4.8** anyway."
- But in this one https://github.com/lxc/lxc/commit/9b5724cd5808f45fd77edd6189a10f199d8f0153, it mentions gnu99 and forbids only **gcc version below 4.6 (excluded)**

--- a/configure.ac
+++ b/configure.ac
@@ -53,7 +53,7 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #define CLANG_VERSION \
 	(__clang_major__ * 10000 + __clang_minor__ * 100 + __clang_patchlevel__)
 
-#if GCC_VERSION < 40600 && CLANG_VERSION < 10000
+#if GCC_VERSION < 40700 && CLANG_VERSION < 10000
 #error Sorry, your compiler is too old - please upgrade it
 #endif
 	]])], [valid_compiler=yes], [valid_compiler=no])
@@ -683,8 +683,9 @@ AC_PROG_SED
 # See if we support thread-local storage.
 LXC_CHECK_TLS
 
+CFLAGS="$CFLAGS -Wvla -std=gnu11"
 if test "x$enable_werror" = "xyes"; then
-	CFLAGS="$CFLAGS -Werror -Wvla -std=gnu11"
+	CFLAGS="$CFLAGS -Werror"
 fi
 
 AC_ARG_ENABLE([thread-safety],

